Cython/Compiler/ModuleNode.py

```diff
@@ -956,9 +958,6 @@ def generate_struct_union_definition(self, entry, code):
                 code.putln("  #pragma pack(push, 1)")
                 code.putln("#endif")
             code.putln(header)
-            var_entries = scope.var_entries
-            if not var_entries:
-                error(entry.pos, "Empty struct or union definition not allowed outside a 'cdef extern from' block")
```

Comment:
Because empty enums are now detected at the parsing stage, I had to move detection of struct/unions to there as well, as they are all tested together.